src/test/java/com/example/webapp/Service/EmployeeServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package com.example.webapp.Service;
+
+
+import com.example.webapp.Exception.EmployeeNotFoundException;
+import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.Repository.EmployeeRepository;
+import com.example.webapp.dto.EmployeeDTO;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static com.example.webapp.util.EMPLOYEE_FOR_CHANGE;
+import static com.example.webapp.util.EMPLOYEE_LIST;
+import static com.example.webapp.util.EMPLOYEE_LIST_DTO;
+import static com.example.webapp.util.NEW_EMPLOYEE;
+import static com.example.webapp.util.NEW_EMPLOYEE_DTO;
+import static com.example.webapp.util.VALID_DEPARTMENT;
+import static com.example.webapp.util.VALID_EMPLOYEE;
+import static com.example.webapp.util.VALID_EMPLOYEE_DTO;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class EmployeeServiceTest {
+    @Mock
```

Comment:
Here you can also use MockBean from Spring Boot Starter test and dependency will be automatically injected